cpp/src/parquet/arrow/arrow_schema_test.cc

```diff
@@ -763,7 +763,7 @@ TEST_F(TestConvertParquetSchema, ParquetSchemaArrowExtensions) {
     props.set_arrow_extensions_enabled(true);
     auto arrow_schema = ::arrow::schema(
         {::arrow::field("json_1", ::arrow::extension::json(), true),
-         ::arrow::field("json_2", ::arrow::extension::json(::arrow::large_utf8()),
+         ::arrow::field("json_2", ::arrow::extension::json(::arrow::utf8()),
```

Comment:
Instead of making the test easier, can you keep `large_utf8` and ensure it passes?

Reply:
Can we reconstruct storage type if it is not stored? In this case we know data is of JSON logical type, but we don't know what was it's storage type at write time. Am I missing something?

Reply:
(There was a test where arrow schema was available and wasn't used at read time. I've addressed that)

Reply:
> Can we reconstruct storage type if it is not stored? In this case we know data is of JSON logical type, but we don't know what was it's storage type at write time. Am I missing something?

No, I'm asking you to write the test so that it reflects that.

Reply:
Something like this? https://github.com/apache/arrow/pull/44215/commits/e522efb7a0d291573e8c44548b68e9d120e70328

Reply:
Yes.